pkg/registry/resolver.go

```diff
@@ -49,7 +51,22 @@ func (r *registry) Resolve(config api.MultiStageTestConfiguration) (types.TestFl
 }
 
 func (r *registry) process(steps []api.TestStep) (internalSteps []types.TestStep, errs []error) {
+	// unroll chains
+	var unrolledSteps []api.TestStep
 	for _, external := range steps {
+		if external.Chain != nil {
+			var err error
+			steps, err := r.unrollChain(external)
```

Comment:
nit: this `steps` shadows the `steps` from the function parameter, let's use a different name for clarity